File: src/app/model/computer-reservation.ts
// src/app/models/computer-reservation.model.ts

export interface ComputerReservation {
    reservation_id: string;
    computer_id: string;
    member_id: string;
    reservation_date: string;
    start_time: string;
    end_time: string;
    status: string;
    created_date: string;
    
    // For display purposes
    computer_name?: string;
    member_name?: string;
    branch_name?: string;
  }
  
  export interface ComputerReservationFilter {
    computer_id?: string;
    member_id?: string;
    date_from?: string;
    date_to?: string;
    status?: string;
    branch_id?: string;
  }

File: LibraryManagement.Core/Entities/ComputerReservation.cs
using System;
using System.Collections.Generic;

namespace LibraryManagement.Core.Entities;

public partial class ComputerReservation
{
    public string ReservationId { get; set; } = null!;

    public string? ComputerId { get; set; }

    public string MemberId { get; set; } = null!;

    public DateOnly ReservationDate { get; set; }

    public TimeOnly StartTime { get; set; }

    public TimeOnly EndTime { get; set; }

    public string? Status { get; set; }

    public DateTime CreatedDate { get; set; }

    public virtual Computer? Computer { get; set; }

    public virtual Member Member { get; set; } = null!;
}
